app/services/trainees/create_from_hesa.rb

```diff
@@ -23,7 +23,7 @@ def initialize(student_node:, record_source:)
 
     def call
       Audited.audit_class.as_user(USERNAME) do
-        trainee.assign_attributes(mapped_attributes)
+        trainee.assign_attributes(mapped_attributes) unless @current_trainee_state == :awarded
```

Comment:
@richardpattinson @markhallen i would have expected this to be a guard clause on the first line of the method:

```
return if @current_trainee_state == :awarded
```
as it stands, you're not assigning attributes but the trainee is still saved and lines 28 to 34 still run. this means that:

- hesa_metadata is updated
- degrees are updated

which I don't think we want to happen.